src/lib/y2network/interface_config_builder.rb

```diff
@@ -91,5 +87,70 @@ def load_sysconfig(devmap)
     def load_s390_config(devmap)
       @s390_config = !devmap.nil? ? @s390_config.merge(devmap) : @s390_config
     end
+
+  private
+
+    # returns a map with device options for newly created item
+    def init_mandatory_options
+      # FIXME: NetHwDetection is done in Lan.Read
+      Yast.import "NetHwDetection"
+
+      # #104494 - always write IPADDR+NETMASK, even empty
+      # #50955 omit computable fields
+      config["BROADCAST"] = ""
+      config["NETWORK"] = ""
+
+      if !config["NETMASK"] || config["NETMASK"].empty?
+        config["NETMASK"] = Yast::NetHwDetection.result["NETMASK"] || "255.255.255.0"
+      end
+      if !config["STARTMODE"] || config["STARTMODE"].empty?
+        config["STARTMODE"] = new_device_startmode
+      end
+      if !config["DHCLIENT_SET_DEFAULT_ROUTE"] || config["DHCLIENT_SET_DEFAULT_ROUTE"].empty?
+        config["DHCLIENT_SET_DEFAULT_ROUTE"] = "no"
+      end
+    end
+
+    # returns default startmode for a new device
+    #
+    # startmode is returned according product, Arch and current device type
+    def new_device_startmode
+      Yast.import "ProductFeatures"
+
+      product_startmode = Yast::ProductFeatures.GetStringFeature(
+        "network",
+        "startmode"
+      )
+
+      startmode = case product_startmode
+      when "ifplugd"
+        if replace_ifplugd?
+          hotplug_usable? ? "hotplug" : "auto"
+        else
+          product_startmode
+        end
+      when "auto"
+        "auto"
+      else
+        hotplug_usable? ? "hotplug" : "auto"
+      end
+
+      startmode
+    end
+
+    def replace_ifplugd?
+      Yast.import "Arch"
+      Yast.import "NetworkService"
+
+      return true if !Yast::Arch.is_laptop
+      return true if Yast::NetworkService.is_network_manager
+      return true if ["bond", "vlan", "br"].include? type
+
+      false
+    end
+
+    def hotplug_usable?
```

Comment:
I am not sure if name is not misleading. From code it looks like if device is hotplug of not, right?

Reply:
yes

Reply:
so what about `hotplug_interface?` as method name? that "usable" is quite confusing for me.